syntax = "proto3";
import "google/protobuf/empty.proto";

package open_greenery.rpc.light;

service LightProxy {
  rpc GetConfig(google.protobuf.Empty) returns (ConfigResponse) {}
  rpc GetManualControl(google.protobuf.Empty) returns (ManualControlResponse) {}
  rpc GetMode(google.protobuf.Empty) returns (ModeResponse) {}
  rpc SetStatus(StatusReport) returns (google.protobuf.Empty) {}
}

message ConfigResponse {
  // msecs since start of day
  optional int32 day_start = 1;
  optional int32 day_end = 2;
}

message ManualControlResponse {
  enum Control {
    CONTROL_ENABLE = 0;
    CONTROL_DISABLE = 1;
    CONTROL_TOGGLE = 2;
  }
  optional Control control = 1;
}

message ModeResponse {
  enum Mode {
    MODE_MANUAL = 0;
    MODE_AUTO = 1;
  }
  optional Mode mode = 1;
}

message StatusReport {
  bool is_enabled = 1;
}
